File: CapsValidationProyect.Persistence/DapperConnection/Pagination/PaginationModel.cs
using System.Collections.Generic;
namespace CapsValidationProyect.Persistence.DapperConnection.Pagination
{
    public class PaginationModel
    {
        public List<IDictionary<string,object>> RecordList {get;set;}
        public int TotalRecords {get;set;}
        public int TotalPages {get;set;}
    }
}

File: CapsValidationProyect.NuxtClient/Entities/PaginationModel.ts
export interface IPaginationModel {
    recordList: Array<{ [key: string]: any }>;
    totalRecords: number;
    totalPages: number;
  }

  export class PaginationModel implements IPaginationModel {
    recordList: Array<{ [key: string]: any }>;
    totalRecords: number;
    totalPages: number;
  
    constructor(
      recordList: Array<{ [key: string]: any }> = [],
      totalRecords: number = 0,
      totalPages: number = 0
    ) {
      this.recordList = recordList;
      this.totalRecords = totalRecords;
      this.totalPages = totalPages;
    }
  }